pkg/resources/fluentd/fluentd.go

```diff
@@ -331,6 +336,21 @@ func (r *Reconciler) reconcileDrain(ctx context.Context) (*reconcile.Result, err
 			continue
 		}
 
+		if inUse && hasJob {
+			pvcLog.Info("deleting drainer job early as PVC is now in use")
+
+			if err := client.IgnoreNotFound(r.Client.Delete(ctx, &job, client.PropagationPolicy(v1.DeletePropagationBackground))); err != nil {
```

Comment:
What would happen if we use foreground deletion here? Why background deletion is more appropriate? I have a sense that it would be better to keep the job there until the pod is completely removed to avoid any race conditions by accident...

Reply:
sorry, it was copy-paste 😅 changed it to force foreground deletion